aiobotocore/tokens.py

```diff
@@ -152,6 +157,14 @@ async def _refresher(self):
             token_dict["accessToken"], expiration=expiration
         )
 
+    @CachedProperty
```

Comment:
I wonder if this is or needs to be reentry safe? I'm afraid the cached async context manager might only be consumed once. Any thoughts?

Reply:
this is a good point.. I guess in botocore the client itself is cached, but here create_nested_client is context manager. Thinking using @property instead of @CachedProperty would solve reentry issue, so we get a fresh context manager each time